doc/api/http.md

```diff
@@ -128,6 +128,11 @@ http.request(options, onResponseCallback);
 added: v0.11.4
 -->
 
+* `options` {Object} Options containing connection details. Check
+  [`net.createConnection()`][] for the format of the options
+* `callback` {Function} Callback function that receives the created socket
+* Returns: {net.Socket}
```

Comment:
The documentation states:

> A socket/stream can be supplied in one of two ways: by returning the socket/stream from this function, or by passing the socket/stream to `callback`.

So it's not a "true" async or sync function. Indeed, the [code](https://github.com/nodejs/node/blob/master/lib/_http_agent.js#L187-L189) explicitly supports both sync and async usage:

```js
  const newSocket = self.createConnection(options, oncreate);
  if (newSocket)
    oncreate(null, newSocket);
```